### PR TITLE
Adding passing of advanced matching parameters to track calls

### PR DESCRIPTION
**Pull request by creativefunction13**
## 💸 TL;DR

Adds the advanced matching parameters to every track call now that the pixel can accept them.  This is so that if the data-layer changes, we use the updated values.

## 📜 Details

[Jira](https://reddit.atlassian.net/browse/ACS-607)
These changes pass along the data-layer information with each `track` call.  This allows for updated information added to the data layer after initial load to be available for later events.

## 🧪 Testing Steps / Validation

Test added to validate the advanced matching parameters added to metadata in track calls.
image

## ✅ Checks

- [x] CI tests (if present) are passing
- [ ] Adheres to code style for repo
- [ ] Contributor License Agreement (CLA) completed if not a Reddit employee

### CODE REVIEW

**Comment by creativefunction13** [comment]
Deployment of this change will need to wait for the next pixel release so that advanced parameters are supported in `track`.

File: template.tpl [hunk 1]
```diff
@@ -1409,8 +1408,45 @@ scenarios:
     // Verify that the tag finished successfully.
     assertApi('makeTableMap').wasCalledWith(mockData.advancedMatchingParams, 'name', 'value');
     assertApi('gtmOnSuccess').wasCalled();
+- name: Test Track Advanced Matching Parameters
```
Comment (achembarpu) [inline template.tpl:1411]: Could we add a test specifically for this flow:
- email/other being empty during init
- but then being set later during track

**Review by achembarpu — CHANGES_REQUESTED** [review]
Looks good! I substituted the staging pixel.js and tested this - appeared to work as expected. I can test again once live.

File: template.tpl (hunk 1 above)
Comment (creativefunction13) [reply template.tpl:1411]: So this test was somewhat doing that, but I made it more explicit.  I also added a second test for the values being set once, and then updated in the second GTM invocation.

Comment (achembarpu) [reply template.tpl:1411]: Perfect, thanks!

**Review by achembarpu — APPROVED** [review]
LGTM! Tested working.

**Review by monicating — APPROVED** [review]

**Pull request merged**